src/ext/service.rs

```diff
@@ -109,23 +127,23 @@ pub struct Sources<'a> {
     /// All `Source` trees available for this repository.
     sources: &'a sources::Sources,
     /// Do we need to iterate over our `context` field?
-    context: Option<(String, dc::Context)>,
+    context: Option<(String, Option<String>, dc::Context)>,
     /// Libraries
     libs: vec::IntoIter<(String, String)>,
 }
 
 impl<'a> Iterator for Sources<'a> {
-    type Item = Result<(String, &'a Source)>;
+    type Item = Result<(String, Option<String>, &'a Source)>;
```

Comment:
...over and over again, which is another argument in favor of replacing this tuple with a named `struct` with `pub` fields.

Reply:
I just started making this change, and if you look very closely, the third part of the tuple is actually different. It's initially a `dc::Context`, and then it becomes a `&Source`. Let me see if I can unify them.